text/0000-wordpress-source-plugins.md

```diff
@@ -0,0 +1,157 @@
+
+Finding the right balance for this is ongoing with ACF: a companion plugin is installed in WordPress to add support for the REST API, but it’s `gatsby-source-wordpress` that’s responsible for translating the dat
+
+
+# Adoption strategy
+
+> If we implement this proposal, how will existing Gatsby developers adopt it? Is
+this a breaking change? Can we write a codemod? Should we coordinate with
+other projects or libraries?
+
+This approach gives us the opportunity to coordinate with the WordPress plugin community, without the requirement of doing so.
+
+The WordPress community has been moving closer towards the JS community already, especially with the most recent release of WordPress:
+
+> Take Every Opportunity to really beef up your JavaScript Chops
+> <footer>—Mall Mullenweg, co-creator of WordPress</footer>
+
+…as summarised by Wes Bos in [<cite>Learn JavaScript</cite>](https://wesbos.com/learn-javascript/)).
+
+Gravity Forms team has written about on [Gutenberg](https://wordpress.org/plugins/gutenberg/) (WordPress’ re-build of their content editor in React, etc, see also https://github.com/gatsbyjs/gatsby/issues/7465):
+
+> Not only is Gutenberg is changing the editing experience, but it'll also change how WordPress plugins are used and developed. … Instead of clicking the Add Form button above the editor to place a shortcode the page, a form can be placed just like any other content by simply adding a block.
+> <footer>—Jeff Matson, Gravity Forms, [<cite>Gutenberg is Coming. Are You Ready?</cite>](https://www.gravityforms.com/gutenberg-coming/)</footer>
+
+With this change, a WordPress plugin authors might opt to maintain their own subplugn as well, as it would provide a clear way of using their REST API implementation. If not, this approach still allows a member of the community who has a direct need for the plugin to maintain the subplugin, and can do so without needing to coordinate with the WordPress plugin author.
+
+# How we teach this
+
+> What names and terminology work best for these concepts and why? How is this
+idea best presented? As a continuation of existing Gatsby patterns?
+> Would the acceptance of this proposal mean the Gatsby documentation must be
+re-organized or altered? Does it change how Gatsby is taught to new developers
+at any level?
+> How should this feature be taught to existing Gatsby developers?
+
+# Unresolved questions
+
+> Optional, but suggested for first drafts. What parts of the design are still
+TBD?
+
+- **[Detailed design](#detailed-design)**
+- **Would this mean moving the existing plugins supported out of `gatsby-source-wordpress`?** Right now I think we should keep ACF support in `gatsby-source-wordpress`, but I can also see the case for moving all WordPress plugin…plugins out of it.
```

Comment:
I think we can actually move this out of main package, but still ship them by default (at least initially). This would also allow us to validate that our extension points actually do make sense if we would implement ACF (and other things we support) in separate packages that use public wordpress plugin extension points